docs/code-guide.md

```diff
@@ -0,0 +1,133 @@
+# Coding Guidelines
+
+This document defines the coding standards for the OCM UI project. All code contributions must follow these guidelines to ensure consistency, maintainability, and code quality.
+
+These guidelines serve to:
+- Establish shared conventions for code structure and patterns
+- Improve code reviews by setting clear expectations
+- Maintain a consistent, maintainable codebase
+
+Use these guidelines during development and code reviews. When in doubt or if a scenario isn't covered, ask the team for help.
+
+## Component Structure
+
+Components should be organized by features. Each feature is represented by a container component in charge of handling logic and rendering other presentational components. The main responsibilities of presentational components are to display data received via props and handle user interactions.
+
+Components should be properly broken down into UI elements responsible for a single goal. 
+
+Example folder structure for a hypothetical `UsersList` component:
+
+```
+UsersList/
+├── UsersList.tsx           # Container component
+├── UsersList.test.tsx      
+├── UsersList.stories.tsx      
+├── useUsersList.ts         # Feature-specific hooks
+├── components/             # Presentational components (only imported by container)
+├── └── UsersTable.tsx      # Only create a folder if you have many
+├── └── UsersToolbar.tsx
+├── types.ts                # Only if needed      
+└── utils/                  # Feature-specific utilities
+```
+
+## Coding Patterns
+
+Use the following patterns when designing components:
+- UI components should be a thin wrapper around data, they should handle local state only when necessary.
+- Try to flatten the UI state into a basic calculation, deriving data from props
+- Create a new component abstraction when you're nesting conditional logic or top-level if/else statements. Ternaries are reserved for small, easily readable logic.
+- When complex data manipulation and logic is necessary, make use of custom hooks.
```

Comment:
I would add sharing logic between different components as well
For example: https://github.com/RedHatInsights/uhc-portal/pull/240/changes/f2592d3e519214d439901278356fb82c6a9477bf